### DISCUSSION TITLE
Optional params check

### DISCUSSION THREAD

**Discussion by kathirsvn** [stargate/stargate-mongoose#22] [General]
Hi @vkarpov15 
Should we check the call back function param as in https://github.com/Automattic/mongoose/blob/7fbb440c4fa2edf878052d717fbf6407c4381142/lib/query.js#L2574 in all of the APIs and reset the optional params received in a function, accordingly? I think last when we discussed you mentioned, it's a nice to have feature. Should we create an issue and track for implementing later?

cc: @amorton @jeffreyscarpenter

**Comment by vkarpov15** [comment]
@kathirsvn I'd advise against it. Mongoose 7 drops support for callbacks, so if you implement Mongoose's "if any argument is a function, use it as a callback" logic you'll have to remove it very soon. It also isn't necessary, because Mongoose only uses 1 function signature for driver calls.

**Reply by kathirsvn** [reply]
Cool. Got it, thanks @vkarpov15
